examples/gltiles-geotiff-tiles.js

```diff
@@ -0,0 +1,140 @@
+// These tiles are 16-bit 4-sample RGB+NIR geotiff
+const tcr = new GlTiledTextureGeoTiffTiles({
+  xyz: tiffTiles,
+  geotiffFactory: GeoTIFF.fromUrl,
+  sample: 0,
+  fillValue: -999,
+  fetchFuncName: 'getTCR',
+  pool: pool,
+});
+const tcg = new GlTiledTextureGeoTiffTiles({
+  xyz: tiffTiles,
+  geotiffFactory: GeoTIFF.fromUrl,
+  sample: 1,
+  fillValue: -999,
+  fetchFuncName: 'getTCG',
+  pool: pool,
+});
+const tcb = new GlTiledTextureGeoTiffTiles({
+  xyz: tiffTiles,
+  geotiffFactory: GeoTIFF.fromUrl,
+  sample: 2,
+  fillValue: -999,
+  fetchFuncName: 'getTCB',
+  pool: pool,
+});
+const nir = new GlTiledTextureGeoTiffTiles({
+  xyz: tiffTiles,
+  geotiffFactory: GeoTIFF.fromUrl,
+  sample: 3,
+  fillValue: -999,
+  fetchFuncName: 'getNIR',
+  pool: pool,
+});
+
+const rgbnirShader =
+  '#line 1                                   \n' +
+  'void main(void) {                                              \n' +
+  '	// Fetch texel values for different bands                     \n' +
+  '	// in the current texel coordinate (vTextureCoords.st)        \n' +
+  '	// These values are in ranges like 0..4000, and must be       \n' +
+  '	// normalized into the 0..1 range for display in GL.          \n' +
+  '	float tcr = getTCR(vTextureCoords.st);                        \n' +
+  '	float tcg = getTCG(vTextureCoords.st);                        \n' +
+  '	float tcb = getTCB(vTextureCoords.st);                        \n' +
+  '	float nir = getNIR(vTextureCoords.st);                        \n' +
+  '                                                               \n' +
+  ' if (tcr < 30.) { gl_FragColor = vec4(0.0); } else              \n' +
+  ' {                                                             \n' +
+  '	gl_FragColor = vec4(                                          \n' +
+  '    tcr / 4000.,                                               \n' +
+  '    tcg / 3000.,                                               \n' +
+  '    tcb / 2000.,                                               \n' +
+  '    1.0);                                                      \n' +
+  ' }                                                             \n' +
+  '}                                                              \n';
+
+const glSource = new GlTiles({
+  fragmentShader: rgbnirShader,
+  textureSources: [tcr, tcg, tcb, nir],
+  attributions:
+    "<a href='https://s2maps.eu'>Sentinel-2 cloudless</a> by <a href='https://eox.at/'>EOX IT Services GmbH</a> (Contains modified Copernicus Sentinel data 2019)",
+  uniforms: {},
+  projection: 'EPSG:4326',
+  tileGrid: tileGrid,
+});
+
+const map = new Map({
+  layers: [
+    new TileLayer({
+      source: new TileWMS({
+        url: 'https://ahocevar.com/geoserver/wms',
+        params: {
+          'LAYERS': 'ne:NE1_HR_LC_SR_W_DR',
+          'TILED': true,
+        },
+      }),
+      opacity: 0.5,
+    }),
+
+    new TileLayer({
+      source: glSource,
+      tileSize: [1024, 1024],
+      tileGrid: tileGrid,
```

Comment:
A TileLayer does not do anything with `tileSize` and `tileGrid` properties.
```suggestion
```

Reply:
My Leaflet brain still struggles with the way OL deals with tile layers and tile grids :confused: